pkg/cluster/fixetcd.go

```diff
@@ -0,0 +1,500 @@
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      jobNameFixPeers,
+					Namespace: nameSpaceEtcds,
+					Labels:    map[string]string{"app": jobNameDataBackup},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: serviceAccountName,
+					Containers: []corev1.Container{
+						{
+							Name:  "backup",
+							Image: image,
+							Command: []string{
+								"/bin/bash",
+								"-cx",
+								backupOrFixEtcd,
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: to.BoolPtr(true),
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "PEER_PODS",
+									Value: peerPods,
+								},
+								{
+									Name:  "DEGRADED_NODE",
+									Value: de.Node,
+								},
+								{
+									Name:  "FIX_PEERS",
+									Value: "true",
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "host",
+									MountPath: "/host",
+									ReadOnly:  false,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{
+		FieldManager: jobNameFixPeers,
+	})
+	if err != nil {
+		return err
+	}
+
+	timeout := adjustTimeout(ctx, log)
+	log.Infof("Waiting for %s", jobNameFixPeers)
+	select {
+	case <-time.After(timeout):
+		log.Infof("Finished waiting for job %s", jobNameFixPeers)
+		break
+	case <-ctx.Done():
+		log.Warnf("Request cancelled while waiting for %s", jobNameFixPeers)
+	}
+
+	log.Infof("Deleting %s now", jobNameFixPeers)
+	propPolicy := metav1.DeletePropagationBackground
+	err = kubecli.BatchV1().Jobs(b.Namespace).Delete(ctx, jobNameFixPeers, metav1.DeleteOptions{
+		PropagationPolicy: &propPolicy,
+	})
+	if err != nil {
+		return err
+	}
+
+	// return errors from deferred delete functions
+	return err
+}
+
+func createPrivilegedServiceAccount(ctx context.Context, log *logrus.Entry, name, usersAccount string, kubecli kubernetes.Interface, securitycli securityv1client.SecurityContextConstraintsInterface) (*securityv1.SecurityContextConstraints, *rbacv1.ClusterRole, *rbacv1.ClusterRoleBinding, *corev1.ServiceAccount, error) {
+	serviceAcc, err := kubecli.CoreV1().ServiceAccounts(nameSpaceEtcds).Create(ctx, &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		AutomountServiceAccountToken: to.BoolPtr(true),
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+
+	cr, err := kubecli.RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: usersAccount,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				// TODO restrict permissions to required
```

Comment:
Is this TODO meant for a future release, or before we go live with this maintenance task?

Reply:
I need to see exactly which permissions are required, I'll do that and update this.

Reply:
Reminder on this TODO.